src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -297,9 +297,10 @@ export class AccountManagementService implements IAccountManagementService {
 	// UI METHODS //////////////////////////////////////////////////////////
 	/**
 	 * Opens the account list dialog
-	 * @return Promise that finishes when the account list dialog opens
+	 * @param waitForDialogClose whether to wait for the dialog to close
+	 * @return Promise that finishes when the account list dialog opens or closes based on the value of waitForDialogClose parameter
 	 */
-	public openAccountListDialog(): Thenable<void> {
+	public openAccountListDialog(waitForDialogClose: boolean): Thenable<void> {
```

Comment:
Hm, I guess this is fine but I'm not a huge fan of using parameters for this kind of thing (changing what the return value means).

What about just returning a nested Thenable, with the internal one being one that resolves when the dialog is closed?

`Thenable<Thenable<void>>`

Reply:
I introduced this parameter to simplify the unit tests, but now i found an easy way to mock the account dialog as well, so this is no longer needed, we don't really have a scenario now that needs this to be conditional, so change it to make it always wait for the dialog close.